connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -491,6 +490,10 @@ private void closePartitions() {
     private class HandleRebalance implements ConsumerRebalanceListener {
         @Override
         public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+            // Increase the commit sequence number so that asynchronous commits that are not complete
+            // won't overwrite what we do here
+            commitSeqno += 1;
+            committing = false;
```

Comment:
These two flags are now modified when a rebalance occurs. Incrementing the `commitSeqno` and setting `committing = false` are necessary to effectively ignore any in-flight async commit handler when it calls `onCommitCompleted`. Without this, the in-flight commit handler will call `onCommitCompleted` that _will_ modify and overwrite the correct `lastCommittedOffsets` with the previous generation offsets.